sql/core/src/test/resources/sql-tests/results/string-functions.sql.out

```diff
@@ -533,39 +533,35 @@ AABB
 
 
 -- !query
-SELECT lpad('abc', 5, x'12')
+SELECT lpad('abc', 5, x'57')
 -- !query schema
-struct<>
+struct<lpad(abc, 5, X'57'):string>
 -- !query output
-org.apache.spark.sql.AnalysisException
-cannot resolve 'lpad('abc', 5, X'12')' due to data type mismatch: Arguments 'str' and 'pad' of function 'lpad' must be the same type.; line 1 pos 7
+WWabc
```

Comment:
@viirya this changes/fixes the behavior of non-ansi (now it's the same with spark 3.2).

Reply:
This restores the behavior of different data types under non-ansi mode. But seems we need to fail under ansi mode? Not seen in this change.

Reply:
What this PR does is: at compile-time, when we see `lpad/rpad` is called with mixed string/binary inputs, we create `StringL/RPad`.

Then ANSI and non-ANSI has different behaviors: ANSI can not implicitly cast binary to string, so the query fails analysis. With non-ANSI mode, we can implicitly cast binary to string, so the result is the same as 3.2 and before.

Reply:
Oh, I see. That is clear now. Make sense. I was confused because I saw in previous PR you did the analysis failing explicitly in `StringL/RPad`. Now we reply on the implicit cast behavior under non-ansi and ansi modes.